app.rb

```diff
@@ -0,0 +1,120 @@
+require_relative 'person'
+require_relative 'book'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+
+class App
+  attr_accessor :person, :books, :rentals
+
+  def initialize()
+    @person = []
+    @books = []
+    @rentals = []
+  end
+
+  def create_book
+    puts ''
+    print 'Enter book title: '
+    title = gets.chomp
+    print 'Enter book author: '
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Success'
+  end
+
+  def list_all_books
+    puts ''
+    @books.each { |book| puts "Title: #{book.title}", "Author: #{book.author}" }
+  end
+
+  def create_person
+    puts ''
+    print 'Enter 1 to add Student and 2 to add Teacher: '
+    selection = gets.chomp.to_i
+    print 'Enter name: '
+    name = gets.chomp
+    print 'Enter age: '
+    age = gets.chomp.to_i
+
+    case selection
+    when 1
+      print 'Does student have parent permission [Y/N]: '
+      permission = gets.chomp.downcase == 'y'
+      student = Student.new(1, age, permission, name)
+      @person << student
+
+    when 2
+      print 'What is the teachers specialization: '
+      specialization = gets.chomp
+      teacher = Teacher.new(age, specialization, name)
+      @person << teacher
+    end
```

Comment:
- [ ] Good job working on this but I get an error anytime I add a teacher and try to view people. Please kindly help fix this.

![image](https://github.com/AbbasSarwar/OOP_school_library/assets/43995830/f649aa10-f7a8-41fd-990d-affdc36d68ec)